presto/utils.py

```diff
@@ -216,88 +180,111 @@ def process_parquet(df: pd.DataFrame) -> pd.DataFrame:
+        # and remove them from the dataset
+        latest_obs_position = df.groupby(["sample_id"])[
+            ["valid_position", "timestamp_ind", "valid_position_diff"]
+        ].max()
+        df["is_last_available_ts"] = (
+            df["sample_id"].map(latest_obs_position["timestamp_ind"]) == df["timestamp_ind"]
         )
-        df = df[~df["sample_id"].isin(samples_before_start_date)]
-        df = df[~df["sample_id"].isin(samples_after_end_date)]
-
-    # add timesteps before the start_date where needed
-    intermediate_dummy_df = pd.DataFrame()
-    for n_ts_to_add in range(1, MIN_EDGE_BUFFER + 1):
-        samples_to_add_ts_before_start = latest_obs_position[
-            (MIN_EDGE_BUFFER - latest_obs_position["valid_position"]) >= -n_ts_to_add
+        samples_after_end_date = latest_obs_position[
+            (latest_obs_position["valid_position"] > latest_obs_position["timestamp_ind"])
         ].index
-        dummy_df = df[
-            (df["sample_id"].isin(samples_to_add_ts_before_start)) & (df["timestamp_ind"] == 0)
-        ].copy()
-        dummy_df["timestamp"] = dummy_df["timestamp"] - pd.DateOffset(
-            months=n_ts_to_add
-        )  # type: ignore
-        dummy_df[feature_columns] = NODATAVALUE
-        intermediate_dummy_df = pd.concat([intermediate_dummy_df, dummy_df])
-    df = pd.concat([df, intermediate_dummy_df])
-
-    # add timesteps after the end_date where needed
-    intermediate_dummy_df = pd.DataFrame()
-    for n_ts_to_add in range(1, MIN_EDGE_BUFFER + 1):
-        samples_to_add_ts_after_end = latest_obs_position[
-            (MIN_EDGE_BUFFER - latest_obs_position["valid_position_diff"]) >= n_ts_to_add
+        samples_before_start_date = latest_obs_position[
+            (latest_obs_position["valid_position"] < 0)
         ].index
-        dummy_df = df[
-            (df["sample_id"].isin(samples_to_add_ts_after_end)) & (df["is_last_available_ts"])
-        ].copy()
-        dummy_df["timestamp"] = dummy_df["timestamp"] + pd.DateOffset(
-            months=n_ts_to_add
-        )  # type: ignore
-        dummy_df[feature_columns] = NODATAVALUE
-        intermediate_dummy_df = pd.concat([intermediate_dummy_df, dummy_df])
-    df = pd.concat([df, intermediate_dummy_df])
-
-    # Now reassign start_date to the minimum timestamp
-    new_start_date = df.groupby(["sample_id"])["timestamp"].min()
-    df["start_date"] = df["sample_id"].map(new_start_date)
 
-    # Also reassign end_date to the maximum timestamp
-    new_end_date = df.groupby(["sample_id"])["timestamp"].max()
-    df["end_date"] = df["sample_id"].map(new_end_date)
-
-    # reinitialize timestep_ind
-    df["timestamp_ind"] = (df["timestamp"].dt.year * 12 + df["timestamp"].dt.month) - (
-        df["start_date"].dt.year * 12 + df["start_date"].dt.month
+        if len(samples_after_end_date) > 0 or len(samples_before_start_date) > 0:
+            logger.warning(
+                f"""\
+    Removing {len(samples_after_end_date)} \
```

Comment:
so in practice this should never happen or it would mean in extraction something went wrong, right? Good sanity check anyway.

Reply:
yes, indeed, in practice it should never happen. but in reality for some Phase I samples this was the case, that's why this check appeared.